astroquery/vo_service_discovery/core.py

```diff
@@ -0,0 +1,245 @@
+        """
+        Query the Virtual Observatory registry to find services which can then be searched.
+
+        Parameters
+        ----------
+        service_type : str, required
+            Valid VO service types are: "conesearch", "simpleimageaccess", "simplespectralaccess", "tableaccess"
+            They may be shortened to "cone", "image", "spectra", "spectrum", or "table" or "tap", respectively.
+        keyword : str, optional, default is None
+            The query will return any services which contain this keyword in their ivoid, title, or description.
+        waveband : str, optional, default is None
+            Comma-delimited list of wavebands.  Results will contain all services that offer at least one of these wavebands.
+            Legal values are listed below.
+        source : str, optional, default is None
+            Any substring in ivoid (a unique identifier of the service)
+        publisher : str, optional, default is None
+            The name of any publishing organization (e.g., "stsci", "heasarc")
+        order_by : str, optional, default is None
+            What column to order the results by.  The returned columns are:
+                "waveband","short_name","ivoid","res_description","access_url","reference_url","publisher", service_type"
+        url : str, optional, default points to the STScI Registry TAP service
+            Base URL for the desired TAP service.
+        output_file : str, optional, default is to use the TAP job ID
+            File name where the results are saved if dumpToFile is True.
+        output_format : str, optional, default 'votable'
+            results format
+        dump_to_file : bool, optional, default is False
+            If True, the results are saved in a file instead of using memory.
+        verbose : bool, optional, default is False
+            When True, the ADQL query computed from the keyword arguments will be printed.
+        """
+
+        query = self._build_adql(service_type, keyword, waveband, source, publisher, order_by)
+        if verbose:
+            print('Registry:  sending query ADQL = {}\n'.format(query))
+
+        aptable = self._sync_tap_query(query, url=url, output_file=output_file, output_format=output_format,
+                                      dump_to_file=dump_to_file, verbose=verbose)
+
+        return aptable
+
+    def query_counts(self, field, minimum=1, url=None, output_file=None, output_format="votable",
+                     dump_to_file=False, verbose=False):
+        """
+        Query the Virtual Observatory registry to find what values exist, and counts for each value,
+        for the 'service_type', 'waveband', or 'publisher' fields in the main query.
+
+        Parameters
+        ----------
+        field : str, required
+            One of 'service_type', 'waveband' or 'publisher'
+        minimum : int, optional, default is 1
```

Comment:
Add an explanation on the next line, like:
```
        Include only those results with at least this many counts.
```

Reply:
Addressed.